roles/power_aix_bootstrap/tasks/yum_install.yml

```diff
@@ -1,98 +1,105 @@
-    delegate_to: localhost
+- name: Check for script file on localhost
+  stat:
+    path: "{{ download_dir }}/yum_installer.sh"
+  register: script_dest
+  delegate_to: localhost
 
-  - name: Copy the installer script file
-    copy:
-      src: "files/yum_installer.sh"
-      dest: "{{ download_dir }}/yum_installer.sh"
-      mode: '0755'
-    when: not script_dest.stat.exists
-    delegate_to: localhost
+- name: Copy the installer script file
+  copy:
+    src: "files/yum_installer.sh"
+    dest: "{{ download_dir }}/yum_installer.sh"
+    mode: '0755'
+  when: not script_dest.stat.exists
+  delegate_to: localhost
 
 # DEFINE / EXPAND target path
-  - name: Check for target directory
-    raw: "test -d {{ target_dir }} && echo true || echo false"
-    register: target_exists
+- name: Check for target directory
+  raw: "test -d {{ target_dir }} && echo true || echo false"
+  register: target_exists
+  changed_when: false
 
-  - name: Check for target yum bundle
-    raw: "test -e {{ target_dir }}/{{ yum_src }} && echo true || echo false"
-    register: yum_target_exists
-    when: target_exists.stdout is search("true")
+- name: Check for target yum bundle
+  raw: "test -e {{ target_dir }}/{{ yum_src }} && echo true || echo false"
+  register: yum_target_exists
+  when: target_exists.stdout is search("true")
 
-  - name: Create target filesystem for image transfer
-    raw: "crfs -v jfs2 -g rootvg -a size=200M -m {{ target_dir }} -A yes -p rw"
-    register: crfs_result
-    when: target_exists.stdout is search("false")
+- name: Create target filesystem for image transfer
+  raw: "crfs -v jfs2 -g rootvg -a size=200M -m {{ target_dir }} -A yes -p rw"
+  register: crfs_result
+  when: target_exists.stdout is search("false")
 
-  - name: Mount target filesystem
-    raw: "mount {{ target_dir }}"
-    when: crfs_result.rc is defined and crfs_result.rc == 0
+- name: Mount target filesystem
+  raw: "mount {{ target_dir }}"
+  when: crfs_result.rc is defined and crfs_result.rc == 0
 
 # TRANSFER files to target inventory
-  - name: Transfer install images
-    raw: "scp -p {{ download_dir }}/yum_installer.sh {{ download_dir }}/{{ rpm_src }} {{ download_dir }}/{{ yum_src }} root@{{ inventory_hostname }}:{{ target_dir }}"
-    register: scp_result
-    ignore_errors: True
-    delegate_to: localhost
+- name: Transfer install images
+  raw: >
+    "scp -p {{ download_dir }}/yum_installer.sh
+    {{ download_dir }}/{{ rpm_src }}
+    {{ download_dir }}/{{ yum_src }}
+    root@{{ inventory_hostname }}:{{ target_dir }}"
+  register: scp_result
+  ignore_errors: true
+  delegate_to: localhost
+  changed_when: scp_result not failed
 
 # EXECUTE restore on target inventory
-  - name: Restore yum bundle content
-    raw: "{{ target_dir }}/yum_installer.sh {{ target_dir }}"
-    ignore_errors: True
-    when: scp_result.rc == 0
+- name: Restore yum bundle content
+  raw: "{{ target_dir }}/yum_installer.sh {{ target_dir }}"
+  failed_when: false
```

Comment:
We need to investigate the reason the playbook was set to ignore the errors

Reply:
i am not sure when about the reason why this task will fail, but it seems that in the original playbook it didn't matter if this task fails or not (i assume, it is related to trying to run this command, but ends up not being needed, so it returns error because there is nothing to do - like having an error when you use `rm` command to a file that does not exist)

so in the role linting CI, ansible prefers to used `failed_when` for cases where you ignore errors instead